### DISCUSSION TITLE
Incompatible released speaker encoder models with current code base

### DISCUSSION THREAD

**Discussion by vince62s** [coqui-ai/TTS#1077] [General Q&A]
Hi,

I got the feeling the speaker encoder models released here: https://github.com/coqui-ai/TTS/wiki/Experimental-Released-Models

are not longer compatible with the current master code base.

Can you confirm ?

Wondering is there is a 256 embedding size model already trained somewhere.

**Comment by erogol** [comment]
I've not tried it for a while, and it is likely incompatible. 

Have you tried loading it?

**Comment by vince62s** [comment]
yes it throws an error.
In this file for instance:
https://github.com/coqui-ai/TTS/blob/main/TTS/speaker_encoder/configs/config.json

we need to change the first line
"model_name": "lstm" ==> "model":"speaker_encoder"
and then below
    "model": {
        "input_dim": 40,
        "proj_dim": 256,
        "lstm_dim": 768,
        "num_lstm_layers": 3,
        "use_lstm_with_projection": true
    },
change "model" to "model_params" and add a subsection "model_name": "lstm"

then it loads, but with the checkpoint from the link above the result is not so good.

Alternative:
modify the register_config function to align speaker_encoder classes to the others (more work)

**Reply by erogol** [reply]
Experimental releases are not intended to be supported in the long term.

For the rest, you can maybe send a PR that fixes the default config.json?
